scene/resources/primitive_meshes.cpp

```diff
@@ -1466,7 +1466,7 @@ void SphereMesh::_create_mesh_array(Array &p_arr) const {
 			} else {
 				Vector3 p = Vector3(x * radius * w, y, z * radius * w);
 				points.push_back(p);
-				Vector3 normal = Vector3(x * radius * w * scale, y / scale, z * radius * w * scale);
+				Vector3 normal = Vector3(x * w * scale, radius * (y / scale), z * w * scale);
 				normals.push_back(normal.normalized());
```

Comment:
Probably a dumb question but why is it `* scale` on `x` and `z`, and `/ scale` on `y`?

Reply:
scale is basically just the "vertical radius", so you can think of a the ellipsoid as a unit sphere scaled by "scale" in the y axis and by "radius" in the x and z axis.

Because of this, the normal actually needs to be scaled in the x and z axis by `scale`, and in the y axis by `radius`. This is a particular case of using the transpose of the inverse of the transform matrix, which one usually does when transforming normals. See my explanation in https://github.com/godotengine/godot/pull/51995.

The `/scale` factor on `y` comes from the fact that `y`is already scaled vertically  (since `y = scale * cos(Math_PI * v);`), so by dividing by scale we get `cos(Math_PI*v)`

To sum up:
If `(x,y,z)` is a point in the unit sphere, the (not normalized) normal of the scaled ellipsoid is given by:
`normal = (x*scale,y*radius,z*scale)`